### ISSUE TITLE
doesn't all the shap values sum to 1 for LSTM model

### ISSUE THREAD

**Issue by akshayuppal3** [shap/shap#605]
what does the sum of values represent in case of LSTM model? I thought the sum of shap values should be equal to the predicted value?

I am using deep explainer and I am training on all of the training sentences and explaining the prediction of a single sentence

train_sample = X_train[np.random.choice(X_train.shape[0], 700, replace=True), :]
explainer = shap.DeepExplainer(model, train_sample)

shap_values = explainer.shap_values(X_test[0:1])

output --
array([-0.11484171,  0.01492828,  0.08478769,  0.18077848,  0.10010771,
        0.12915095,  0.07681657,  0.05064181,  0.04806488, -0.04686701,
        0.00179007,  0.00092119,  0.00042222])

The predicted value is 0.99 but the sum is not equal to that 

I am using the same model architecture that you used for IMDB movie review

max_len = max_len
model = Sequential()
model.add(Embedding(vocalb, dimension, input_length=max_len, trainable=False)) 
model.add(LSTM(units=100, return_sequences=True))
model.add(Flatten())
model.add(Dense(100, activation='relu'))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])

Thanks

**Comment by detrin** [comment]
@akshayuppal3 The sum of all SHAP values will be equal to E[f(x)] — f(x).
https://christophm.github.io/interpretable-ml-book/shapley.html

@thatlittleboy CFC

**Comment by thatlittleboy** [comment]
Actually it's `sum(shap_values) = f(x) - E[f(x)]`.

**Comment by detrin** [comment]
> Actually it's `sum(shap_values) = f(x) - E[f(x)]`.

thanks, good to know
